### ISSUE TITLE
Add tracing support to async-stripe

### ISSUE THREAD

**Issue by arlyon** [arlyon/async-stripe#300]
### Is your feature request related to a problem? Please describe.

It would be useful for observability reasons to instrument and log useful information about async-stripe such as spans and certain internal state such as the status of the retry state machine etc.

### Describe the solution you'd like

Add tracing instrumentation and logging where appropriate.

**Comment by seanpianka** [comment]
You mentioned in my original PR that tracing instrumentation was intentionally not added. What was the motivation for that? 

I think it would be fair to add `tracing::debug` and `#[tracing::instrument]` to some of the network-related functions (e.g. anything in the `client` module seems suitable for this).

**Comment by arlyon** [comment]
At the time when I added it to the codegen there was some msrv problem that caused compile problems. I suppose we could have used a lower version but decided to leave it. That problem has since been resolved I believe :)

**Comment by arlyon** [comment]
- [ ] Trace during the request / response process

**Comment by augustoccesar** [comment]
@arlyon if interested, when using [Octocrab](https://github.com/XAMPPRocky/octocrab) it integrates quite well with [OpenTelemetry](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client-span) and have `tracing` as a feature flag. Could be some source of inspiration.

https://github.com/XAMPPRocky/octocrab/blob/c3169920cb1961ee269f0c307b43d6e42e81d809/src/lib.rs#L735-L784

**Comment by arlyon** [comment]
Yeah I am a huge fan of tracing and use it extensively. Plan is to add tracing only to the clients themselves (there is actually nothing stopping you from creating a client now in your own crate that impl's the trait), instrumenting execute, and annotating IO with some additional data such as retries etc. looking forward to getting it out there

**Comment by arlyon** [comment]
I have started this, I will get a PR up for tracing in the client calls as well

**Comment by arlyon** [comment]
@claude please add tracing::instrument to the core api calls
